pkg/services/grafana-apiserver/common.go

```diff
@@ -42,16 +42,12 @@ type APIRouteHandler struct {
 	Handler http.HandlerFunc // when Level = resource, the resource will be available in context
 }
 
-// APIRoutes define the
+// APIRoutes define explicit HTTP handlers in an apiserver
+// TBD: is this actually necessary -- there may be more k8s native options for this
```

Comment:
I'll try a few more services... but we may not actually need these.

Reply:
> http://localhost:3000/apis/example.grafana.app/v0alpha1/aaa (custom root route)
> http://localhost:3000/apis/example.grafana.app/v0alpha1/bbb (custom root route)
> http://localhost:3000/apis/example.grafana.app/v0alpha1/namespaces/default/ccc (namespace level custom route)

i think `rest.Connecter` could be used for these as well, and [`rest.Scoper`](https://pkg.go.dev/k8s.io/apiserver@v0.27.1/pkg/registry/rest#Scoper) would control if it is cluster or namespaced scoped. `/aaa` and `/bbb` can also be added with `NonGoRestfulMux`, but i don't think we get API discovery or openapi for free in that approach.

Reply:
Yes -- lets explore some more, when @charandas and I tried this earlier the problem is that we could not have both restful and `NonGoRestfulMux` alive in the same service at the same time.  The non-resful flavor goes first and captures all events

Reply:
> restful and NonGoRestfulMux alive in the same service at the same time

by service, do you mean API group? if so, that makes sense. i forgot i was testing with apiextensions-apiserver, which uses restful for the CRD API group, but not for the custom resource routes